src/main/java/webserver/JsonTodos.java

```diff
@@ -50,19 +48,24 @@ public void addTodo(JSONObject newTodo) {
         JSONArray todos = getAllTodos();
         todos.add(newTodo);
 
-        try (FileWriter file = new FileWriter(todosFile)) {
+        try (FileWriter file = fileUtils.fileWriter(todosFile)) {
             file.write(todos.toJSONString()); 
             file.flush();
         } catch (IOException e) {
             e.printStackTrace();
         }
     }
 
-    public void updateTodo(JSONObject updatedTodo) {
+    public void updateTodo(int id, boolean isDone) {
         JSONArray todos = getAllTodos();
-        todos.add(updatedTodo);
-
-        try (FileWriter file = new FileWriter(todosFile)) {
+        try (FileWriter file = fileUtils.fileWriter(todosFile)) {
+            for (JSONObject todo: (Iterable<JSONObject>) todos) {
```

Comment:
I've seen this pattern a few times, where you have to cast.  Usually (but not always), this can be avoided, along with the possible consequence of getting a `ClassCastException`.  

A good stepping stone might be to add a method to `JSONArray` that does the casting for you.

```java
class JSONArray {
  ...
  public Iterable<JSONObject> asIterable() {
    //KDK: totally guessing here
    return this.todoItems.stream()
      .map(x -> JSONObject.class.cast(x))
      .collect(Collectors.toList());
  }
  ...
}
```